api/app/helpers/lightning.ts

```diff
@@ -29,11 +30,49 @@ const payInvoice = async (invoice: string) => {
       fee_limit_sat: FEE_LIMIT_SAT,
     });
     if (res.status === 200) {
-      return { success: true, data: res.data };
+      const data = res.data;
+      const responseJsonStrings = data
+        .split("\n")
+        .filter((str: string) => str.trim() !== "");
+      const jsonObjects = responseJsonStrings.map((str: string) =>
+        JSON.parse(str)
+      );
+      const jsonArray = jsonObjects.map(
+        (obj: { result: PayInvoiceResponse }) => obj.result
+      );
+      // loop through the json array and check if the payment preimage is not
+      // "0000000000000000000000000000000000000000000000000000000000000000"
+      // check the status of the objects in the filtered array if its "SUCCEEDED"
+      const unsuccessfulPreimage =
+        "0000000000000000000000000000000000000000000000000000000000000000";
+      const filteredArray = jsonArray.filter(
+        (obj: PayInvoiceResponse) =>
+          obj.payment_preimage !== unsuccessfulPreimage &&
+          obj.status === "SUCCEEDED"
+      ) as PayInvoiceResponse[];
+      if (filteredArray.length > 0) {
+        Logger.info(`Payment successful: ${filteredArray[0].payment_preimage}`);
+        return {
+          success: true,
+          data: filteredArray,
+          error: null,
+        };
+      } else {
+        Logger.error(
+          `Payment failed for ${jsonArray[0].payment_hash} with reason: ${
+            jsonArray[jsonArray.length - 1].failure_reason
+          }`
+        );
+        return {
+          success: false,
+          data: null,
+          error: new Error("Payment failed"),
+        };
+      }
     }
   } catch (err) {
     console.error(err);
```

Comment:
Add a logger for the error including the invoice.

From line38, `JSON.parse(str)` could throw errors for non-standard JSON strings while the payment could have been successful. I was thinking of try/catch but working with malformed objects doesn't seem like a good idea.

```suggestion
    Logger.error(
          `Payment failed for invoice:${invoice} 
          err: ${err?.message ?? err}`
        );
```

Reply:
Done!

Reply:
Nice! You can include the Lightning Invoice in the logger. 
i.e `payment failed for invoice: ${invoice} with err…`

also no need for console.error anymore

Reply:
Yup! Shows like this in the logs.
```
error:   Payment failed for invoice: lnbc60u1pntyn5tpp5vkupeh4wwe65sa03cgvkyvxnsxj3dty062ut8s0zy06yshpy6qzqdpgw3jhxapqvf6xxgr5wfskuumrwf5hqapqd3shyem9cqzzsxqyz5vqsp5cwj0a2wl2vgu0d4knvklmq6lqq4vukxrqm4k3lw6s2javp0aq72q9qxpqysgqvqzt5d3uc5m33hf4u9k4z4tp5nd7lddldwmaf7acsnwenr3srxwxspdwghwnt2sltr4z3h23aht7786rlp93kmdwf2k92g32sl43l4cqadr0qq
Error: Request failed with status code 409
    at /Users/<user>/bitcoinsearch/transcription-review-backend/api/app/controllers/lightning.controller.ts:95:13
    at Generator.next (<anonymous>)
    at fulfilled (/Users/<user>/bitcoinsearch/transcription-review-backend/api/app/controllers/lightning.controller.ts:5:58)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```